oauth.go

```diff
@@ -494,19 +501,50 @@ func (c *Consumer) makeAuthorizedRequest(method string, url string, dataLocation
 	sort.Sort(paramPairs)
 
 	queryParams := ""
-	separator := "?"
-	if dataLocation == LOC_BODY {
-		separator = ""
+	separator := ""
+	contentType := ""
+	switch dataLocation {
+	case LOC_URL:
+		separator = "?"
+	case LOC_BODY:
+		contentType = "application/x-www-form-urlencoded"
+	case LOC_MULTIPART:
+		pipeReader, pipeWriter := io.Pipe()
+		writer := multipart.NewWriter(pipeWriter)
+		go func(body io.Reader) {
+			part, err := writer.CreateFormFile(multipartName, "/no/matter")
```

Comment:
If you don't want to specify a filename, it looks like you can call writer.CreateFormField() and omit the filename

I couldn't tell from reading the twitter docs whether this would work for sure, but it looks like it should.